### ISSUE TITLE
fish_postexec and fish_prompt don't emit when command is aborted (e.g. with ctrl-c)

### ISSUE THREAD

**Issue by CGamesPlay** [fish-shell/fish-shell#7320]
This is an exact duplicate of #2356, which has regressed, however that issue is locked.

```
fish, version 3.1.2
Darwin Ryans-Macbook.local 18.7.0 Darwin Kernel Version 18.7.0: Sun Dec  1 18:59:03 PST 2019; root:xnu-4903.278.19~1/RELEASE_X86_64 x86_64
TERM=xterm-256color
```

```fish
function postexec --on-event fish_postexec
    echo postexec
end
function prompt --on-event fish_prompt
    echo prompt
end
sleep 300 # Press ctrl-c to abort
```

If you run this code and press `^C` within 5 minutes, you should expect to see the text "postexec" and "prompt". Instead, you see nothing.

**Comment by zanchey** [comment]
The fix for #2356 is not in 3.1.2, but only in the (yet to be released, I am getting there) 3.2.0.

**Comment by CGamesPlay** [comment]
Oh, my mistake. Thanks for the clarification; can't wait for the release!
